### PR TITLE
Z-test

### LINKED ISSUE #5: Z-test

### PR DESCRIPTION
**Pull request by hamzakpt**

### CODE REVIEW

**Review by zakariabeni — CHANGES_REQUESTED** [review]
HI, 
Actually, I expect more about Z-test because it is what we face in the first time learning about normal distribution. I have some suggestions.
- link to colab.
- you can put some equation related to how to calculated Z.
- put Z table and how to refer the solution in the table.
- display or visualize normal distribution graph based on data.
- in sample task, you used N = 10, but in conclusion you mentioned that Z-test is used for sample size greater than 30. So, it is kind of contradiction.
- you can expand the steps like: after a step there will be a piece of code or explanation. It would be attractive to read.

Thank you.

File: z-test.ipynb [hunk 1]
```diff
@@ -0,0 +1,757 @@
...
... valid or repeatable. For example, if someone said they had found a new drug that cures cancer, you would want to be sure it was probably true. A hypothesis test will tell you if it’s probably true, or probably not true. A Z test, is used when your data is approximately normally distributed. A Z-test is any statistical test for which the distribution of the test statistic under the null hypothesis can be approximated by a normal distribution. Z-test tests the mean of a distribution in which we already know the population variance σ2 . Because of the central limit theorem, many test statistics are approximately normally distributed for large samples. For each significance level in the confidence interval, the Z-test has a single critical value (for example, 1.96 for 5% two tailed) which makes it more convenient than the Student's t-test which has separate and different critical values for each sample size (for different sample size, it would have different degree of freedom, which may determine the value of the critical values). Therefore, many statistical tests can be conveniently performed as approximate Z-tests if the sample size is large or the population variance is known. \n",
```
Comment (apoyezzhayev) [inline z-test.ipynb]: "Hypothesis testing is just a way for you to figure out if results from a test are valid or repeatable." - it's not precise definition.

Comment (apoyezzhayev) [inline z-test.ipynb]: "Z-test tests the mean of a distribution in which we already know the population variance σ2 " - what does it mean? It's better to say that we test equality of some value to population's mean. Almost always some value is **sample** distribution's mean.

File: z-test.ipynb [hunk 2]
```diff
@@ -0,0 +1,757 @@
...
+     "output_type": "stream",
+     "text": [
+      "mean of data:  88.44713820775404 , sigma of data: 3.0397185090150947\n"
+     ]
+    }
+   ],
+   "source": [
+    "points = wine_data['points']\n",
+    "data_mean = points.mean()\n",
+    "data_sigma = points.std(ddof=0)\n",
```
Comment (apoyezzhayev) [inline z-test.ipynb]: Add a comment that usage of ddof=0 is important and why

**Review by apoyezzhayev — CHANGES_REQUESTED** [review]
Besides comments from review of @zakariabeni 
1) Use latex for math formulas to avoid:
   - x_bar != (not equal) μ.
   - H0
   - σ/sqrt(N), etc.
2) Add or at least describe different use cases for single-tailed tests μ≥μ0 or μ≤μ0
3) At the end of the notebook add one-line z-test from one of python statistical libraries
4) May be add some comments about variance estimation calculation because cases when we definitely know the population's variance are uncommon. It's a nuisance parameters for the test.
5) Add some visualizations for:
   - t-test and z-test comparisons
   - Normal distribution graph with colored tails
6) Add comments that besides one-sample location test there are other applications of Z-test and add references to them
7) Look at some review-comments inside notebook

**Review by sergey-jr — CHANGES_REQUESTED** [review]
I think same as @apoyezzhayev and @zakariabeni.

**Comment by nikitml** [comment]
hey @hamzakpt , it's "action required", please take a look

**Review by sergey-jr — CHANGES_REQUESTED** [review]
Seems like nothing is changed.
- no interaction
- no tasks, questions
- no formulas of Z-test (how to calculate)
- no vizualization
- you don't use latex for equations (that you used)

Overall picture is bad

**Review by zakariabeni — CHANGES_REQUESTED** [review]
Seem nothing changes, please refer to my first comment.

**Review by zakariabeni — APPROVED** [review]

**Review by apoyezzhayev — APPROVED** [review]

**Review by sergey-jr — CHANGES_REQUESTED** [review]
Pros:
- formula for z-score
- fixed equation with latex

Cons:
- no interaction at all
- no tasks and self-questions
- no link to colab
- just one example

**Comment by hamzakpt** [comment]
> Pros:
> 
> * formula for z-score
> * fixed equation with latex
> 
> Cons:
> 
> * no interaction at all
> * no tasks and self-questions
> * no link to colab
> * just one example

included self questions, and provide him colab link. and Interaction provided.

**Review by sergey-jr — APPROVED** [review]

**Pull request merged**